pci/src/device.rs

```diff
@@ -41,7 +43,16 @@ impl Display for Error {
 pub trait PciDevice: BusDevice {
     /// Assign a legacy PCI IRQ to this device.
     /// The device may write to `irq_evt` to trigger an interrupt.
-    fn assign_irq(&mut self, _irq_evt: EventFd, _irq_num: u32, _irq_pin: PciInterruptPin) {}
+    fn assign_irq(
+        &mut self,
+        _irq_evt: Arc<Box<Fn(u16) + Send + Sync>>,
```

Comment:
Would the `Fn(u16)` closure parameter be an IRQ number for non MSI/PCI based transports?

Reply:
Well in case of IRQ, we will simply ignore it. The point of this `u16` is to select based on the right Queue, but in case of IRQs, there's only one IRQ for all the queues from a device.

Reply:
Nevermind, I changed it so that it takes no parameter at all.